docs/design/design.md

```diff
@@ -49,11 +49,52 @@ This means we aim to standardize:
      - `kube-scheduler.yaml`
  - `/etc/kubernetes/kubelet.conf` as the path where the kubelet should store its credentials to the API server.
  - `/etc/kubernetes/admin.conf` as the path from where the admin can fetch his/her superuser credentials.
+ - Names of certificates files:
+   - `ca.crt`, `ca.key`
+   - `apiserver.crt`, `apiserver.key`
+   - `apiserver-kubelet-client.crt`, `apiserver-kubelet-client.key`
+   - `sa.pub`, `sa.key`
+   - `front-proxy-ca.crt`, `front-proxy-ca.key`
+   - `front-proxy-client.crt`, `front-proxy-client.key`
 
 
 ## `kubeadm init` phases
 
-### Phase 1: Generate the necessary certificates
+`kubeadm init` internal workflow consists of a sequence of phases.
+
+The `Kubeadm phase` command, alpha in 1.8, allows users to invoke individually each phase, and ultimately offers a reusable and composable API/toolbox that can be used by other kubernetes bootstrap tools / by any IT automation tool / by advanced user for creating custom clusters. 
+
+### Preflight checks
+
+`kubeadm` executes a set of preflight checks before starting the init, with the aim to verify preconditions and avoid common cluster startup problems. 
+In any case the user can skip preflight checks with the `--skip-preflight-checks` option.
+
+- [warning] if kubeadm version "not safe" for installing the required kubernetes version (kubeadm is considered safe for installing patch versions "newest" than kubeadm version itself, not safe for installing "newest" major/minor releases)
```

Comment:
we should reword: "not safe" can mean security-wise, which is not the case.

suggest: [warning] If the Kubernetes version to use (passed with the `--kubernetes-version` flag) is one minor version higher than the kubeadm CLI version, kubeadm will warn the user that it might or might not work. Please upgrade your kubeadm client before starting such a cluster.